### PR TITLE
[YUNIKORN-1842] improve updatePVCRefCounts

### PR DESCRIPTION
**Pull request by FrankYang0529**
### What is this PR for?

In the previous version, we call `updatePVCRefCounts` in `updateNode`, `updatePod`, and `removePod`. These functions call `NodeInfo.SetNode`, `NodeInfo.AddPod`, or `NodeInfo.RemovePod`. All these `NodeInfo` functions update `Generation`, so I think we don't need to use `pvcGeneration` to determine whether we need to update `pvcRefCounts`. We can just give `nodeInfo` object to `updatePVCRefCounts` and update `pvcRefCounts` in `SchedulerCache`.

### What type of PR is it?
* [ ] - Bug Fix
* [X] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?

https://issues.apache.org/jira/browse/YUNIKORN-1842

### How should this be tested?

Add unit tests for `updatePVCRefCounts`

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/cache/external/scheduler_cache.go [hunk 1]
```diff
@@ -603,46 +603,24 @@ func (cache *SchedulerCache) IsPVCUsedByPods(key string) bool {
...
 }
 
-func (cache *SchedulerCache) maxNodeGeneration() int64 {
-	var maxGeneration int64
-	for _, node := range cache.nodesMap {
-		if maxGeneration < node.Generation {
-			maxGeneration = node.Generation
+func (cache *SchedulerCache) updatePVCRefCounts(node *framework.NodeInfo, removeNode bool) {
+	for k, v := range cache.pvcRefCounts {
+		delete(v, node.Node().Name)
```
Comment (craigcondit) [inline pkg/cache/external/scheduler_cache.go:608]: Save node.Node().Name to a variable at the top of the function so it can be re-used.

**Review by craigcondit — CHANGES_REQUESTED** [review]
Minor nit - avoid calling node.Node().Name multiple times.

File: pkg/cache/external/scheduler_cache.go (hunk 1 above)
Comment (FrankYang0529) [reply pkg/cache/external/scheduler_cache.go]: Updated it. Thank you!

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**